lib/codecs/adts.js

```diff
@@ -99,12 +103,10 @@ AdtsStream = function(handlePartialSegments) {
       adtsFrameDuration = (sampleCount * ONE_SECOND_IN_TS) /
         ADTS_SAMPLING_FREQUENCIES[(buffer[i + 2] & 0x3c) >>> 2];
 
-      frameEnd = i + frameLength;
-
-      // If we don't have enough data to actually finish this ADTS frame, return
-      // and wait for more data
-      if (buffer.byteLength < frameEnd) {
-        return;
+      // If we don't have enough data to actually finish this ADTS frame,
+      // then we have to wait for more data
+      if ((buffer.byteLength - i) < frameLength) {
```

Comment:
We should probably move this check up to just after determining the frame length (before the sample count) to prevent errors addressing `i + 6`

Reply:
changed the while loop condition to `i + 7` so that we won't even run the loop if we don't have enough bytes. That way and we can keep all the parsing logic together.